sw/device/tests/crypto/aes_sideload_functest.c

```diff
@@ -0,0 +1,245 @@
+
+  // Initialize entropy complex, which the key manager uses to clear sideloaded
+  // keys. The `keymgr_testutils_startup` function restarts the device, so this
+  // should happen afterwards.
+  return entropy_complex_init();
+}
+
+/**
+ * Test that a sideloaded AES key can be used.
+ *
+ * Loads a sideloaded key, then runs encryption and decryption using the key.
+ */
+status_t basic_encrypt_decrypt_test(void) {
+  // Encrypt the test plaintext.
+  uint32_t ciphertext1[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(encrypt(kKeySalt1, ciphertext1));
+
+  // Decrypt the ciphertext.
+  uint32_t recovered_plaintext[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(decrypt(kKeySalt1, ciphertext1, recovered_plaintext));
+
+  // Check that the recovered plaintext matches the original plaintext.
+  TRY_CHECK_ARRAYS_EQ(recovered_plaintext, kAesPlaintextBlock,
+                      ARRAYSIZE(kAesPlaintextBlock));
+
+  return OK_STATUS();
+}
+
+/**
+ * Test that sideloaded AES keys are updated as expected.
+ *
+ * Different keys should produce different ciphertexts, and the same key
+ * should produce the same ciphertext.
+ */
+status_t sideload_update_test(void) {
+  // Sideload the first key and encrypt the plaintext.
+  uint32_t ciphertext1[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(encrypt(kKeySalt1, ciphertext1));
+
+  // Sideload the second key and encrypt the plaintext.
+  uint32_t ciphertext2[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(encrypt(kKeySalt2, ciphertext2));
+
+  // Check that the ciphertexts are different.
+  TRY_CHECK_ARRAYS_NE(ciphertext1, ciphertext2, ARRAYSIZE(ciphertext1));
+
+  // Sideload the first key again and encrypt the plaintext.
+  uint32_t ciphertext3[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(encrypt(kKeySalt1, ciphertext3));
+
+  // Check that the ciphertexts are the same.
+  TRY_CHECK_ARRAYS_EQ(ciphertext1, ciphertext3, ARRAYSIZE(ciphertext1));
+
+  return OK_STATUS();
+}
+
+/**
+ * Test that sideloaded AES keys are properly cleared.
+ *
+ * Clearing a key should result in an invalid key error.
+ */
+status_t sideload_clear_test(void) {
+  // Sideload the first key and encrypt the plaintext.
+  uint32_t ciphertext1[ARRAYSIZE(kAesPlaintextBlock)];
+  TRY(encrypt(kKeySalt1, ciphertext1));
+
+  // Construct a mock key for the AES driver.
+  uint32_t share0[256 / 32] = {0};
+  uint32_t share1[256 / 32] = {0};
+  aes_key_t key = {
+      .mode = kAesCipherModeCtr,
+      .sideload = kHardenedBoolTrue,
+      .key_len = kAesKeyConfig.key_length,
+      .key_shares = {share0, share1},
+  };
+
+  // Construct the IV.
+  aes_block_t iv;
+  memcpy(iv.data, kAesIv, sizeof(kAesIv));
+
+  // Try to start another encryption operation using the AES driver directly,
+  // and expect an error because the sideloaded key is not present.
+  TRY_CHECK(!status_ok(aes_encrypt_begin(key, &iv)));
+
+  return OK_STATUS();
+}
+
+OTTF_DEFINE_TEST_CONFIG();
+
+bool test_main(void) {
+  static status_t result;
+
+  EXECUTE_TEST(result, test_setup);
```

Comment:
nit: probably should call this inside a `CHECK_STATUS_OK` macro instead of a `EXECUTE_TEST` macro, since it is not a test per se, rather just test setup code?